### PR TITLE
React to pipelines changes

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: samples/SampleApp/Startup.cs [hunk 1]
```diff
@@ -76,7 +76,7 @@ public static Task Main(string[] args)
                     });
 
                     // Run callbacks on the transport thread
-                    options.ApplicationSchedulingMode = SchedulingMode.Inline;
+                    //options.ApplicationSchedulingMode = SchedulingMode.Inline;
```
Comment (davidfowl) [inline samples/SampleApp/Startup.cs]: ?

File: samples/SampleApp/Startup.cs [hunk 2]
```diff
@@ -130,7 +130,7 @@ public static Task Main(string[] args)
                 .UseContentRoot(Directory.GetCurrentDirectory())
                 .UseStartup<Startup>();
 
-            if (string.Equals(Process.GetCurrentProcess().Id.ToString(), Environment.GetEnvironmentVariable("LISTEN_PID")))
+            //if (string.Equals(Process.GetCurrentProcess().Id.ToString(), Environment.GetEnvironmentVariable("LISTEN_PID")))
```
Comment (davidfowl) [inline samples/SampleApp/Startup.cs]: ?

**Comment by pakrym** [comment]
:up::date:

File: src/Kestrel.Core/Internal/Http/Http1MessageBody.cs [hunk 3]
```diff
@@ -62,19 +63,19 @@ protected Http1MessageBody(Http1Connection context)
 
                         if (!readableBuffer.IsEmpty)
                         {
-                            var writableBuffer = _context.RequestBodyPipe.Writer.Alloc(1);
                             bool done;
 
                             try
                             {
-                                done = Read(readableBuffer, writableBuffer, out consumed, out examined);
+                                _context.RequestBodyPipe.Writer.GetMemory();
```
Comment (davidfowl) [inline src/Kestrel.Core/Internal/Http/Http1MessageBody.cs]: Is this the same?

File: tools/CodeGenerator/CodeGenerator.csproj [hunk 4]
```diff
@@ -7,9 +7,6 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
   </PropertyGroup>
 
-  <ItemGroup>
-    <ProjectReference Include="..\..\src\Kestrel.Core\Kestrel.Core.csproj" />
-  </ItemGroup>
```
Comment (halter73) [inline tools/CodeGenerator/CodeGenerator.csproj]: :tada: #1809

File: src/Kestrel.Core/Internal/Http/HttpParser.cs [hunk 5]
```diff
@@ -279,16 +280,20 @@ public unsafe bool ParseHeaders(TRequestHandler handler, ReadOnlyBuffer buffer,
                             else
                             {
                                 var current = reader.Position;
+                                var currentSlice = buffer.Slice(current, bufferEnd);
```
Comment (davidfowl) [inline src/Kestrel.Core/Internal/Http/HttpParser.cs]: @pakrym Won't this performance? This is why I didn't understand why we were removing seek that just took cursors. This means we end up slicing more than we need to.

File: test/Kestrel.Core.Tests/HttpParserTests.cs [hunk 6]
```diff
@@ -384,16 +384,16 @@ public void ExceptionDetailNotIncludedWhenLogLevelInformationNotEnabled()
             Assert.Single(pairs);
             Assert.Equal(headerName, pairs[0].Key);
             Assert.Equal(expectedHeaderValue, pairs[0].Value);
-            Assert.Equal(buffer.End, consumed);
-            Assert.Equal(buffer.End, examined);
+            Assert.True(buffer.Slice(consumed).IsEmpty);
```
Comment (davidfowl) [inline test/Kestrel.Core.Tests/HttpParserTests.cs]: This is pretty bad.

Comment (pakrym) [reply test/Kestrel.Core.Tests/HttpParserTests.cs]: We can add `ROB.PositionsEqual(p1, p2)`

**Comment by davidfowl** [comment]
Are you meeting this into dev?

File: src/Kestrel.Core/Internal/Http/HttpParser.cs (hunk 5 above)
Comment (davidfowl) [reply src/Kestrel.Core/Internal/Http/HttpParser.cs]: Hopefully we don’t regress performance any more than we already have

**Comment by pakrym** [comment]
Yes, I need to prepare other repos first, versions would be different.

**Pull request merged**